src/main/java/fi/metatavu/dcfb/server/items/ItemController.java

```diff
@@ -119,29 +118,34 @@ public Item findItem(UUID itemId) {
    *
-   * @param title title
-   * @param description description
-   * @param category category
-   * @param visibilityLimited visibilityLimited
-   * @param location location
-   * @param slug slug
-   * @param expiresAt expiresAt
-   * @param unitPrice unitPrice
-   * @param priceCurrency priceCurrency
-   * @param amount amount
-   * @param unit unit
-   * @param soldAmount soldAmount
+   * @param title                      title
+   * @param description                description
+   * @param category                   category
+   * @param visibilityLimited          visibilityLimited
+   * @param location                   location
+   * @param slug                       slug
+   * @param expiresAt                  expiresAt
+   * @param unitPrice                  unitPrice
+   * @param priceCurrency              priceCurrency
+   * @param amount                     amount
+   * @param unit                       unit
+   * @param soldAmount                 soldAmount
    * @param allowPurchaseContactSeller allowPurchaseContactSeller
-   * @param allowPurchaseCreditCard allowPurchaseCreditCard
-   * @param deliveryTime deliveryTime
-   * @param contactEmail contactEmail
-   * @param contactPhone contactPhone
-   * @param termsOfDelivery termsOfDelivery
-   * @param sellerId sellerId
-   * @param modifier modifier
+   * @param allowPurchaseCreditCard    allowPurchaseCreditCard
+   * @param deliveryTime               deliveryTime
+   * @param contactEmail               contactEmail
+   * @param contactPhone               contactPhone
+   * @param termsOfDelivery            termsOfDelivery
+   * @param allowDelivery              allow delivery
+   * @param allowPickup                allow pick up
+   * @param deliveryPrice              delivery price
+   * @param deliveryCurrency           delivery currency
+   * @param sellerId                   sellerId
+   * @param modifier                   modifier
+
    * @return updated item
    */
-  public Item updateItem(Item item, LocalizedEntry title, LocalizedEntry description, Category category, boolean visibilityLimited, Location location, String slug, OffsetDateTime expiresAt, String unitPrice, Currency priceCurrency, Long amount, String unit, Long soldAmount, Boolean allowPurchaseContactSeller, Boolean allowPurchaseCreditCard, Integer deliveryTime, String contactEmail, String contactPhone, String termsOfDelivery, UUID sellerId, UUID modifier) {
+  public Item updateItem(Item item, LocalizedEntry title, LocalizedEntry description, Category category, boolean visibilityLimited, Location location, String slug, OffsetDateTime expiresAt, String unitPrice, Currency priceCurrency, Long amount, String unit, Long soldAmount, Boolean allowPurchaseContactSeller, Boolean allowPurchaseCreditCard, Integer deliveryTime, String contactEmail, String contactPhone, String termsOfDelivery,Boolean allowDelivery, Boolean allowPickup, String deliveryPrice, Currency deliveryCurrency, UUID sellerId, UUID modifier) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Method has 25 parameters, which is greater than 7 authorized. ([squid:S00107](https://sonarcloud.io/organizations/metatavu-oy/rules#rule_key=squid%3AS00107))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=fi.metatavu.dcfb-api%3Adcfb-api&pullRequest=39&issues=AWXYx1xJlgbYnohYXER3&open=AWXYx1xJlgbYnohYXER3)

[ISSUE_KEY]: AWXYx1xJlgbYnohYXER3